src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java

```diff
@@ -13,12 +30,416 @@
+        {
+            return InventoryUtils.isPickaxeInTileEntity(worker.getInventoryCitizen(), buildingToDeliver.getNeededPickaxeLevel());
+        }
+        return InventoryUtils.isToolInTileEntity(worker.getInventoryCitizen(), requiredTool);
+    }
+
+    /**
+     * Gather item from chest.
+     * Gathers only one stack of the item.
+     * @param buildingToDeliver building to deliver to.
+     */
+    private void gatherItems(@NotNull final AbstractBuilding buildingToDeliver)
+    {
+        BlockPos position;
+        if(itemsToDeliver.isEmpty())
+        {
+            final String tool = buildingToDeliver.getRequiredTool();
+            position = wareHouse.getTileEntity().getPositionOfChestWithTool(tool, Utils.PICKAXE.equals(tool) ? buildingToDeliver.getNeededPickaxeLevel() : -1);
+        }
+        else
+        {
+            final ItemStack stack = itemsToDeliver.get(0);
+            position = wareHouse.getTileEntity().getPositionOfChestWithItemStack(stack);
+        }
+
+        if(position == null)
+        {
+            return;
+        }
+
+        if(!worker.isWorkerAtSiteWithMove(position, MIN_DISTANCE_TO_CHEST))
+        {
+            return;
+        }
+
+        final TileEntity tileEntity = world.getTileEntity(position);
+        if(tileEntity instanceof TileEntityChest)
+        {
+            if (itemsToDeliver.isEmpty())
+            {
+                isToolInTileEntity((TileEntityChest) tileEntity, buildingToDeliver.getRequiredTool());
+            }
+            else
+            {
+                final ItemStack stack = itemsToDeliver.get(0);
+                if(isInTileEntity((TileEntityChest) tileEntity, stack))
+                {
+                    itemsToDeliver.remove(0);
+                }
+            }
+        }
+    }
+
+
+    /**
+     * Check the wareHouse for the next task.
+     * @return the next AiState to go to.
+     */
+    private AIState checkWareHouse()
+    {
+        if(!worker.isWorkerAtSiteWithMove(wareHouse.getLocation(), MIN_DISTANCE_TO_WAREHOUSE))
+        {
+            return START_WORKING;
+        }
+
+        @Nullable final AbstractBuilding buildingToDeliver = wareHouse.getTileEntity().getTask();
+        if(buildingToDeliver == null)
+        {
+            return GATHERING;
+        }
+
+        final AbstractBuildingWorker ownBuilding = getOwnBuilding();
+        if(ownBuilding instanceof BuildingDeliveryman)
+        {
+            ((BuildingDeliveryman) ownBuilding).setBuildingToDeliver(buildingToDeliver);
+        }
+
+        return PREPARE_DELIVERY;
+    }
+
+    /**
+     * Check if the deliveryman code should be executed.
+     * More concretely if he has a warehouse to work at.
+     * @return false if should continue as planned.
+     */
+    private boolean checkIfExecute()
+    {
+        worker.setAIMoveSpeed((float) (worker.BASE_MOVEMENT_SPEED + worker.BASE_MOVEMENT_SPEED * worker.getLevel()/WALKING_SPEED_MULTIPLIER));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2209)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2209)